sched.go

```diff
@@ -85,20 +90,25 @@ func (as *scheduler) run() {
 	for !as.close {
 
 		if len(as.ready) == 0 { // (1)
+			as.logger.Trace("No ready tasks, going to sleep")
 			as.signal.Wait() // (2)
-			continue         // (3)
+			as.logger.Trace("Woken up from sleep", zap.Int("ready tasks", len(as.ready)))
+			continue // (3)
 		}
 
 		taskToRun := as.ready[0]
 		as.ready[0] = task{}    // Cleanup any object references reachable from the closure of the task
 		as.ready = as.ready[1:] // (4)
 
-		as.lock.Unlock()    // (5)
+		as.lock.Unlock() // (5)
+		as.logger.Debug("Running task", zap.Int("remaining ready tasks", len(as.ready)))
```

Comment:
This access of `as.ready` is racy.

Reply:
yeah it is, thanks for spotting this

Reply:
Fixed [here](https://github.com/ava-labs/Simplex/pull/58/commits/85a08af4f05dabb90bcf78bb988fcbc9c23812f4).